### PR TITLE
Allow lists in secrets.yaml

### PR DESCRIPTION
**Pull request by danielwelch**
Use case discussed here: https://github.com/danielwelch/hassio-zigbee2mqtt/issues/259

### CODE REVIEW

File: hassio/secrets.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ class SecretsManager(CoreSysAttributes):
     def __init__(self, coresys: CoreSys):
         """Initialize secret manager."""
         self.coresys: CoreSys = coresys
-        self.secrets: Dict[str, Union[bool, float, int, str]] = {}
+        self.secrets: Dict[str, Union[bool, float, int, str, List[int]] = {}
```
Comment (frenck) [inline hassio/secrets.py]: Why is this only a list of integers?

Comment (danielwelch) [reply hassio/secrets.py]: No reason whatsoever, other than that’s all that my add-on’s specific use case would require. I’m happy to expand it to whatever you all deem acceptable.

**Comment by pvizeli** [comment]
You need also to address the CI issues. As next, the current implementation on `addon/validate.py` does support only single types. If you want list support, you need to extend the validator to work correctly with this.

**Comment by danielwelch** [comment]
I guess my proposal would be to move this: https://github.com/home-assistant/hassio/blob/f5c171e44fd8dd4333b7a94b9b846a707401eb90/hassio/addons/validate.py#L359-L364

out of `_single_validate`, and instead do something like this

```python
def validate(struct):
        """Create schema validator for add-ons options."""
        options = {}

        # read options
        for key, value in struct.items():
            # Ignore unknown options / remove from list
            if key not in raw_schema:
                _LOGGER.warning("Unknown options %s", key)
                continue

           # Lookup secret
           if str(value).startswith("!secret "):
               secret: str = value.partition(" ")[2]
               value = coresys.secrets.get(secret)
           if value is None:
               raise vol.Invalid(f"Unknown secret {secret}")

            typ = raw_schema[key]
            try:
                if isinstance(typ, list):
                    # nested value list
                    options[key] = _nested_validate_list(coresys, typ[0], value, key)
                elif isinstance(typ, dict):
                    # nested value dict
                    options[key] = _nested_validate_dict(coresys, typ, value, key)
                else:
                    # normal value
                    options[key] = _single_validate(coresys, typ, value, key)
            except (IndexError, KeyError):
                raise vol.Invalid(f"Type error for {key}") from None

        _check_missing_options(raw_schema, options, "root")
        return options

    return validate
```

**Comment by pvizeli** [comment]
Could work, please add some tests to verify that it works. Thanks

File: hassio/secrets.py [hunk 2]
```diff
@@ -55,7 +57,9 @@ def get(self, secret: str) -> Union[bool, float, int, str]:
 
             # Filter to only get supported values
             self.secrets = {
-                k: v for k, v in data.items() if isinstance(v, (bool, float, int, str))
+                k: v
+                for k, v in data.items()
+                if isinstance(v, (bool, float, int, str, list))
```
Comment (frenck) [inline hassio/secrets.py]: The contents of lists are not filtered/checked this way...

Comment (pvizeli) [reply hassio/secrets.py]: Since we support complex types and use schemes validation, we don't need the type check here?

**Comment by pvizeli** [comment]
Any future plan on this?

**Pull request closed without merging**